src/components/ui/Modal/Modal.tsx

```
@@ -63,11 +62,66 @@ export function Modal({
   onClose,
 }: ModalProps) {
   const titleId = useId()
+  const modalId = useId()
   const panelRef = useRef<HTMLDivElement>(null)
   const previousFocusRef = useRef<HTMLElement | null>(null)
+  const onCloseRef = useRef(onClose)
+  onCloseRef.current = onClose
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/ui/Modal/Modal.tsx:69:3
  67 |   const previousFocusRef = useRef<HTMLElement | null>(null)
  68 |   const onCloseRef = useRef(onClose)
> 69 |   onCloseRef.current = onClose
     |   ^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  70 |   const actionsRef = useRef<ModalAction[]>([])
  71 |
  72 |   const resolvedActions =

(react-hooks/refs)